### PR TITLE
add yalc and scripts at root

### PR DESCRIPTION
**Pull request by clairesunstudio**
Fixed:
  - project: Assets, React, Patternlab, Site
    component: CircleCI
    description: Added yalc to allow linking to local mayflower-assets and mayflower-react dependencies without publishing the public. (#1153)
    issue: DDP-19695
    impact: Patch

Added:
  - project: Assets, React, Patternlab, Site
    component: Docs
    description: Added useful commands and documentation from the root. (#1153)
    issue: DDP-19695
    impact: Patch

### CODE REVIEW

**Review by smurrayatwork — CHANGES_REQUESTED** [review]
Yalc is meant to be a tool for testing or developing WIP packages locally and should not be used as a replacement for the concept of workspaces or other techniques. I'm also not so sure that symlinks work so well with CircleCI in the manner that this PR is currently implemented.

File: .circleci/config.yml [hunk 1]
```diff
@@ -56,10 +65,15 @@ jobs:
...
+      - run:
+          name: Yalc link
+          command: npm run yalc-link:patternlab
+      - run:
+          name: Build
+          command: cd patternlab/styleguide && node_modules/.bin/gulp patternlab:build
       - persist_to_workspace:
-          root: /var/www/code
-          paths: ["*"]
+          root: ~/repo
```
Comment (clairesunstudio) [inline .circleci/config.yml]:
```suggestion
          root: ~/repo/patternlab
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -181,7 +198,7 @@ jobs:
           command: cd react && npm run build-storybook
       - persist_to_workspace:
           root: ~/repo
```
Comment (clairesunstudio) [inline .circleci/config.yml]:
```suggestion
          root: ~/repo/react
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -318,26 +331,30 @@ jobs:
...
+          paths: [site/node_modules]
+      - run:
+         name: Yalc link
+         command: npm run yalc-link:site
       - run:
           name: Validate Package Build
-          command: cd site && yarn build
+          command: cd site && npm run build
       - persist_to_workspace:
           root: ~/repo
```
Comment (clairesunstudio) [inline .circleci/config.yml]:
```suggestion
          root: ~/repo/site
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -318,26 +331,30 @@ jobs:
...
-          paths: ["*"]
+          paths: ["site"]
```
Comment (clairesunstudio) [inline .circleci/config.yml]:
```suggestion
          paths: ["*"]
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -181,7 +198,7 @@ jobs:
           command: cd react && npm run build-storybook
       - persist_to_workspace:
           root: ~/repo
-          paths: ["*"]
+          paths: ["react"]
```
Comment (clairesunstudio) [inline .circleci/config.yml]:
```suggestion
          paths: ["*"]
```

File: .circleci/config.yml [hunk 6]
```diff
@@ -56,10 +65,15 @@ jobs:
...
+          name: Yalc link
+          command: npm run yalc-link:patternlab
+      - run:
+          name: Build
+          command: cd patternlab/styleguide && node_modules/.bin/gulp patternlab:build
       - persist_to_workspace:
-          root: /var/www/code
-          paths: ["*"]
+          root: ~/repo
+          paths: ["patternlab"]
```
Comment (clairesunstudio) [inline .circleci/config.yml]:
```suggestion
          paths: ["*"]
```

File: .circleci/config.yml [hunk 7]
```diff
@@ -24,22 +32,23 @@ backstop_defaults: &backstop_defaults
     - image: circleci/python:2.7.14
 
 jobs:
+  #
   build:
     <<: *patternlab_defaults
```
Comment (clairesunstudio) [inline .circleci/config.yml]:
```suggestion
```

**Pull request closed without merging**